docs/pages/project/contributing/contributing-ui-tests.md

```diff
@@ -133,7 +133,65 @@ In the last step go to ui folder,
 cd ui;
 ```
 
-And run this command for running the test cases:
+## Run the test cases
+
+There are several options we can use to run the test cases, in CLI:
+
+To run playwright UI mode using the browser, you can add `--ui` in the cli, for example:
+```bash
+npx playwright test --ui
+```
+
+If you are using playwright from docker, you can use `--ui-port=<playwright-docker-server>`, for example:
+```bash
+npx playwright test --ui-port=8080
+```
+
+To run playwright for specific project only, for example meshery-provider, you can run this command:
+```bash
+npx playwright test --ui --project=chromium-meshery-provider --ui 
+```
+
+To run specific test, you can add the test file location, for example:
 ```bash
-PW_TEST_CONNECT_WS_ENDPOINT=ws://localhost:8080/ npx playwright test
-```
\ No newline at end of file
+npx playwright test --ui --project=chromium-meshery-provider --ui tests/e2e/service-mesh-performance.spec.js 
+```
+
+For more detail, you can read the [Playwright Cli docs](https://playwright.dev/docs/test-cli)
+
+
+## Testing Meshery & Local 
+
+By default our test cases is running against both Meshery and Local Provider, we are utilizing playwright feature such as:
+
+- StorageState: In meshery [setup auth](https://github.com/meshery/meshery/blob/master/ui/tests/e2e/auth.setup.js), we have 2 storage state, which store a session for both Meshery and Local provider.
+- Project: After the setup completed it will run project based test which depends on storage state for one for Local Provider and one for Meshery Provider
+- Test Parameterize: In Local provider we are limiting some feature, to test against missing feature, we leverage this playwright feature to check or even skipping the test is not possible to run, To do the check you need to specify `provider` directly from the test, and make sure the test is wrapped using Project fixture:
```

Comment:
In meshery [setup auth](https://github.com/meshery/meshery/blob/master/ui/tests/e2e/auth.setup.js), we have 2 storage **states**, which store a session for Meshery and Local providers.
- Project: After the setup **completes** it will **the** run **project-based** test **depending** on **which** storage state for one for Local Provider and one for Meshery Provider
- Test Parameterize: In Local provider we are limiting some **features** to test against. **For the missing features,** we leverage this playwright feature to check or even **skip** the test i**f it is not possible to run**, To do the check you need to specify `provider` directly from the test, and make sure the test is wrapped using: